### PR TITLE
deprecation of _indices

### PR DESCRIPTION
**Pull request by sjsrey**
xref #67

### CODE REVIEW

File: inequality/_indices.py [hunk 1]
```diff
@@ -11,6 +11,9 @@
      :cite:`nijkamp2015cultural`
      https://www.econstor.eu/bitstream/10419/107568/1/dp8782.pdf
 """
+import warnings
+
+warnings.warn("The module _indices is deprecated and will be removed on 2025-01-01.", DeprecationWarning)
```
Comment (jGaboardi) [inline inequality/_indices.py:16]:
```suggestion
warnings.warn(
    "The module _indices is deprecated and will be removed on 2025-01-01.",
    DeprecationWarning,
    stacklevel=2
)
```

**Comment by jGaboardi** [comment]
Should we disable doctests for the `_indices.py` module to get CI green?

File: inequality/_indices.py [hunk 2]
```diff
@@ -11,6 +11,13 @@
      :cite:`nijkamp2015cultural`
      https://www.econstor.eu/bitstream/10419/107568/1/dp8782.pdf
 """
+import warnings
+
+warnings.warn(
+    "The module _indices is deprecated and will be removed on 2025-01-01.",
+    DeprecationWarning,
```
Comment (martinfleis) [inline inequality/_indices.py:18]: DeprecationWarning assumes that we cut a release, then change it to FutureWarning and make another release with that, then enforce the deprecation. Shall we rather go straight for FutureWarning?

Comment (jGaboardi) [reply inequality/_indices.py:18]: Good point, let's change to a `FutureWarning`.

Comment (jGaboardi) [inline inequality/_indices.py:18]:
```suggestion
    FutureWarning,
```

**Review by jGaboardi — APPROVED** [review]

**Comment by jGaboardi** [comment]
@sjsrey shall we pick this back up again to get CI green?

**Comment by jGaboardi** [comment]
> Should we disable doctests for the `_indices.py` module to get CI green?

We still need to address these failures, either by skipping or updating.

**Comment by martinfleis** [comment]
@jGaboardi that is fixed by https://github.com/geopandas/geopandas/pull/3288

**Comment by sjsrey** [comment]
The remaining failure is related to [codecov](https://github.com/pysal/inequality/actions/runs/10396045345/job/28789402671?pr=69#step:5:16) on macos-13. 

@jGaboardi any suggestions?

**Comment by jGaboardi** [comment]
pre-commit.ci run

**Comment by jGaboardi** [comment]
pre-commit.ci autofix

**Comment by jGaboardi** [comment]
The bot is having trouble auto fixing the import sorting for some reason. Running `ruff check inequality --fix` locally should resolve the issue.

**Comment by jGaboardi** [comment]
pre-commit.ci autofix

**Review by jGaboardi — APPROVED** [review]

**Pull request merged**